File: fp/fp.rb
numbers = [1,2,3]

p "filter method"
p numbers.select{|e| e > 1}

p "map method"
p numbers.map{|e| e.to_s}
p numbers.map(&:to_s)

p "map method - hash type list with symbol key"
p [{name:'A'},{name:'B'},{name:'C'}].map{|e| e[:name]}

p "map method - hash type list with string key"
p [{'name'=>'A'},{'name'=>'B'},{'name'=>'C'}].map{|e| e['name']}

p "reduce method"
p numbers.reduce(0){|sum, indv| sum + indv} 
p numbers.reduce(&:+)

File: fp/fp.py

import functools

numbers = [1,2,3]
print("filter method")
print(list(filter(lambda e: e > 1, numbers)))

print("map method")
print(list(map(lambda e: str(e), numbers)))

print("list comprehension")
print([str(e) for e in numbers])

print("map method - dict type list")
d = [{"name": "A"}, {"name": "B"}, {"name": "C"}]
print(list(map(lambda e: e["name"], d)))
print([e["name"] for e in d])

print("reduce method")
print(functools.reduce(lambda a, b: a + b, numbers))
